std/path.d

```diff
@@ -3269,7 +3269,7 @@ body
                             /* Match for:
                              *   pattern[pi0..pi-1] ~ pattern[piRemain..$]
                              */
-                            if (pattmp.ptr == null)
+                            if (() @trusted { return pattmp.ptr; }() == null)
```

Comment:
Let's use this as a forcing function for that. `e1 is e2` should be `@safe ` for any two slice expressions `e1` and `e2`.